platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -258,6 +258,25 @@ - (id)mgl_jsonExpressionObject {
     return [self valueForKeyPath:@"mgl_jsonExpressionObject"];
 }
 
+- (id)mgl_case {
+    NSArray *caseArray = self;
+    NSExpression *falseExpression;
+    
+    if (caseArray.count % 2) {
+        falseExpression = caseArray.lastObject;
+        caseArray = [caseArray subarrayWithRange:NSMakeRange(0, caseArray.count - 1)];
+    }
+    
+    for (int index = 0; index < caseArray.count; index++) {
```

Comment:
`-[NSArray count]` returns an `NSUInteger`, so to avoid the risk of overflow, `index` should be an `NSUInteger`.